### PR TITLE
Merged in rqt changes + 2.8 fix

### PR DESCRIPTION
**Pull request by RoseWall**

### CODE REVIEW

File: conman_ros/src/ros_interface_service.cpp [hunk 1]
```diff
@@ -34,10 +34,13 @@ ROSInterfaceService::ROSInterfaceService(RTT::TaskContext* owner) :
   }
 
   // Connect operation callers
-  RTT::log(RTT::Debug) << "Connecting conamn_ros operation callers..." << RTT::endlog();
-  getBlocks = scheme->getOperation("getBlocks");
-  getGroups = scheme->getOperation("getGroups");
-  switchBlocks = scheme->getOperation("switchBlocks");
+  RTT::log(RTT::Debug) << "Connecting conamn_ros operation callers..." << RTT::endlog(); 
+  getBlocks = RTT::OperationCaller<bool(std::vector<std::string>&, std::vector<std::string>&, bool, bool)>(
+      scheme->getOperation("getBlocks"), scheme->engine());
```
Comment (jbohren) [inline conman_ros/src/ros_interface_service.cpp]: The signature for this `OperationCaller` isn't right, it needs to be the same signature as the `Scheme::getBlocks` function.

File: conman_ros/src/ros_interface_service.cpp [hunk 2]
```diff
@@ -34,10 +34,13 @@ ROSInterfaceService::ROSInterfaceService(RTT::TaskContext* owner) :
...
+  getGroups = RTT::OperationCaller<bool(std::vector<std::string>&, std::vector<std::string>&, bool, bool)>(
+      scheme->getOperation("getGroups"), scheme->engine());
```
Comment (jbohren) [inline conman_ros/src/ros_interface_service.cpp]: The signature for this `OperationCaller` isn't right, it needs to be the same signature as the `Scheme::getGroups` function.

File: conman_ros/src/ros_interface_service.cpp [hunk 3]
```diff
@@ -34,10 +34,13 @@ ROSInterfaceService::ROSInterfaceService(RTT::TaskContext* owner) :
...
 
   // Connect operation callers
-  RTT::log(RTT::Debug) << "Connecting conamn_ros operation callers..." << RTT::endlog();
-  getBlocks = scheme->getOperation("getBlocks");
-  getGroups = scheme->getOperation("getGroups");
-  switchBlocks = scheme->getOperation("switchBlocks");
+  RTT::log(RTT::Debug) << "Connecting conamn_ros operation callers..." << RTT::endlog(); 
+  getBlocks = RTT::OperationCaller<bool(std::vector<std::string>&)>(
+      scheme->getOperation("getBlocks"), scheme->engine());
+  getGroups = RTT::OperationCaller<bool()>(
```
Comment (jbohren) [inline conman_ros/src/ros_interface_service.cpp]: Shouldn't this also be `RTT::OperationCaller<bool(std::vector<std::string>&)>`?

**Comment by RoseWall** [comment]
https://github.com/jbohren/conman/blob/master/conman/include/conman/scheme.h#L94 I think it doesn't take nay parameter, were you thinking of hasGroup?

**Comment by jbohren** [comment]
> https://github.com/jbohren/conman/blob/master/conman/include/conman/scheme.h#L94 I think it doesn't take any parameters, were you thinking of hasGroup? Or am I just really off perhaps?

Ah it's neither, it should be `RTT::OperationCaller<std::vector<std::string>()>`

**Pull request merged**